crates/compiler/parse/src/expr.rs

```diff
@@ -209,39 +209,74 @@ fn loc_term_or_underscore<'a>(
     options: ExprParseOptions,
 ) -> impl Parser<'a, Loc<Expr<'a>>, EExpr<'a>> {
     one_of!(
```

Comment:
The `loc_term_or_underscore` name doesn't make much sense after the refactor anymore. What about just calling it `loc_term_or_closure`, because that's literally what it is now?